src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -533,6 +535,30 @@ public void SolutionBindingUpdated_WhenSetBoundProject_EventsRaisedInExpectedOrd
             }
         }
 
+        [TestMethod]
+        public void GitRepoUpdated_SolutionBingingUpdatedInvoked()
+        {
+            var gitEventsMonitor = new Mock<IGitEvents>();
+
+            // Arrange
+            using (var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker, loggerMock.Object, gitEventsMonitor.Object))
+            {
+                var eventCounter = new EventCounter(testSubject);
+
+                // Act
+                gitEventsMonitor.Raise(x => x.HeadChanged += null, null, null);
+
+                // Assert
+                eventCounter.PreSolutionBindingUpdatedCount.Should().Be(0);
+                eventCounter.SolutionBindingUpdatedCount.Should().Be(1);
+                eventCounter.PreSolutionBindingChangedCount.Should().Be(0);
+                eventCounter.SolutionBindingChangedCount.Should().Be(0);
+
+                eventCounter.RaisedEventNames.Should().HaveCount(1);
+                eventCounter.RaisedEventNames[0].Should().Be("SolutionBindingUpdated");
```

Comment:
The reason for checking the event names is to make sure that the events are raised in the expected order. It's not necessary if there is only one event - lines 552-555 already check that expected events have been raised.

However, why isn't the `Pre` event being raised too?

Reply:
I have realised that and fixed it later PR (stateful branch)

Reply:
Update: I see you've done this as part of `ui/stateful branch mapper`.